src/client.ts

```diff
@@ -718,11 +725,28 @@ export class SpvWalletClient {
   /**
    * Confirm a contact request
    *
-   * @param {string} paymail Contact paymail to modify
-   * @return {void}
+   * @param {string} passcode - The passcode for the contact
+   * @param contact
+   * @param {string} paymail Contact paymail
+   * @param {number} period - The period for the TOTP
+   * @param {number} digits - The number of digits for the TOTP
+   * @returns {Promise<void>}
+   * @throws {ErrorWrongTOTP} If the TOTP is invalid
+   * @throws {ErrorNoXPrivToValidateTOTP} If the xPriv is not set
    */
-  async ConfirmContact(paymail: string): Promise<void> {
-    return await this.http.request(`contact/confirmed/${paymail}`, 'PATCH');
+  ConfirmContact(
+    passcode: string,
+    contact: Contact,
+    paymail: string,
+    period: number,
+    digits: number,
+  ): Promise<boolean> {
+    const isTotpValid = this.ValidateTotpForContact(contact, passcode, paymail, period, digits);
+    if (!isTotpValid) {
+      throw new ErrorWrongTOTP();
+    }
+
+    return this.http.request(`contact/confirmed/${paymail}`, 'PATCH');
```

Comment:
Are you sure this function call is not asynchronous like before?

Reply:
ahh, ok it returns a Promise, so it doesn't need to await

Reply:
ℹ️ Added